framework/wazuh/core/config/client.py

```diff
@@ -0,0 +1,133 @@
+import yaml
+import os
+from typing import Optional
+
+from wazuh.core.common import WAZUH_SERVER_YML
+from wazuh.core.config.models.server import ServerInternalConfig
+from wazuh.core.config.models.central_config import (Config, CommsAPIConfig,
+                                                     ManagementAPIConfig, ServerConfig,
+                                                     IndexerConfig, EngineConfig)
+
+
+class CentralizedConfig:
+    """Class to manage centralized configuration loading and access.
+
+    Attributes
+    ----------
+    _config : Optional[Config]
+        The loaded configuration object, initialized as None.
+    """
+
+    _config: Optional[Config] = None
+
+    @classmethod
+    def load(cls):
+        """Load the configuration from the YAML file.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the configuration file does not exist.
+        """
+        if cls._config is None:
+            if not os.path.exists(WAZUH_SERVER_YML):
+                raise FileNotFoundError(f"Configuration file not found: {WAZUH_SERVER_YML}")
+            with open(WAZUH_SERVER_YML, 'r') as file:
+                config_data = yaml.safe_load(file)
+                cls._config = Config(**config_data)
+
+    @classmethod
+    def get_comms_api_config(cls) -> CommsAPIConfig:
+        """Retrieve the communications API configuration.
+
+        Returns
+        -------
+        CommsAPIConfig
+            The communications API configuration.
+
+        Loads the configuration if it has not been loaded yet.
```

Comment:
Same change applies for the other methods

```suggestion
        """Retrieve the communications API configuration.
        
        Loads the configuration if it has not been loaded yet.

        Returns
        -------
        CommsAPIConfig
            The communications API configuration.
```

Reply:
Changed in e4e54327c2b3a45cbe77a6f9c30cfd19dc9ce0b6.